### ISSUE TITLE
Azure Pipelines affected tests jobs are being canceled

### ISSUE THREAD

**Issue by zcorpan** [web-platform-tests/wpt#14860]
In https://github.com/web-platform-tests/wpt/pull/14824 I made a [commit](https://github.com/web-platform-tests/wpt/pull/14824/commits/af40482c9f2c1f6105658a3001bc523f8a05bc87) that failed [the Azure Pipelines build](https://github.com/web-platform-tests/wpt/runs/50120087).

I asked @foolip for advice on irc:

> foolip zcorpan: I've never seen that before, seems like agents can get restarted while running a job. Can you point John Jansen at it to ask?

@thejohnjansen can you take a look? I suppose restarting the job could work but I wanted to make sure this gets due attention if something needs fixing.

**Comment by foolip** [comment]
I've found this is happening more widely and wasn't just an infra restart or similar. https://github.com/web-platform-tests/wpt/pull/14850 is the earliest PR (in creation order) that I can see is affected, but not every PR since then is affected. https://github.com/web-platform-tests/wpt/pull/14863 is the most recent.

When it fails, it's in the "Update manifest" step. https://gist.github.com/foolip/edb0f8dd21738b0d890a15a5e6afc1d0 has the log for one failure.

I have seen these manifest failures on Azure Pipelines before, but then it's fallen back to generating the manifest. I suspect that the machines are exceeding the GitHub API quote for unauthenticated requests. How that can lead to the job being canceled I don't know, perhaps there is an exit code that Azure Pipelines treats as a request to cancel?

@thejohnjansen @mustjab any pointers appreciated.

**Comment by foolip** [comment]
@jgraham have you ever observed a GitHub API quota issues with this code on Taskcluster, either for upstream wpt or Mozilla's CI?

**Comment by foolip** [comment]
I've sent https://github.com/web-platform-tests/wpt/pull/14865 which should avoid the issue at the cost of making jobs slower.

**Comment by foolip** [comment]
A problem here is that if https://github.com/web-platform-tests/wpt/pull/14865 is merged and it takes a certain amount of jobs to exceed the quota, then it would be very hard to know when it's safe to revert it.

**Comment by foolip** [comment]
That https://github.com/web-platform-tests/wpt/pull/14865 didn't fix the issue suggests that the problem is while generating the manifest from scratch. That's probably an OOM issue, and it would be fixed by fixing the manifest download. @Hexcles we had a similar problem on Taskcluster, right? Can Taskcluster now survive the manifest download failing, or would it also OOM?

**Comment by Hexcles** [comment]
Regarding memory, we changed the memory settings on Taskcluster and it now guarantees at least 8GB of memory which should be enough even if the manifest needs to be regenerated (but not necessarily so if there are bugs in the code that loads multiple copies of the manifest like #14537 that I just fixed). Realistically I think we need 4GB *available* memory, so perhaps 8GB total memory given that macOS itself also needs quite some juice.

I was going to suggest porting my #14290 to Azure, but on a second thought, it doesn't work on macOS at all.
